### PR TITLE
♻️ Extract recommendation item logic from surrounding view

### PR DESCRIPTION
**Pull request by matthiasrohmer**
In preparation of further work on #4666. Rendering the various details will introduce a little more logic. To not further add up to the RecommendationsView I extracted the logic for the item into a separate file while simplifying things a little.

Blocked by #4742.

### CODE REVIEW

File: pixi/src/ui/recommendations/RecommendationItem.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    );
+
+    this.body = element.querySelector('.ap-m-pixi-recommendations-item-body');
+    this.body.id = `body-${content.id}`;
+
+    this.bodyText = element.querySelector(
+      '.ap-m-pixi-recommendations-item-body-text'
+    );
+
+    this.nextButton = element.querySelector('a');
```
Comment (caroqliu) [inline pixi/src/ui/recommendations/RecommendationItem.js:47]: Could this incorrectly get anchor links in side the recommendation body copy?

File: pixi/src/ui/recommendations/RecommendationItem.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  get next() {
+    return this._next;
+  }
+
+  set hidden(hidden) {
+    this.element.hidden = hidden;
+  }
+
+  onClickHeader() {
+    this.expanded = true;
```
Comment (caroqliu) [inline pixi/src/ui/recommendations/RecommendationItem.js:132]:
```suggestion
    this.expanded = !this.expanded;
```

**Review by caroqliu — APPROVED** [review]

File: pixi/src/ui/recommendations/RecommendationItem.js (hunk 2 above)
Comment (matthiasrohmer) [reply pixi/src/ui/recommendations/RecommendationItem.js]: Ugh, thanks...

File: pixi/src/ui/recommendations/RecommendationItem.js (hunk 1 above)
Comment (matthiasrohmer) [reply pixi/src/ui/recommendations/RecommendationItem.js]: Yep, true. Moved this above the rendering of the body.

**Comment by matthiasrohmer** [comment]
This is now ready, PTAL @caroqliu 🙂 

Bildschirmfoto 2020-09-29 um 17 19 39
Bildschirmfoto 2020-09-29 um 17 19 56

File: pages/content/pixi/index.md [hunk 3]
```diff
@@ -66,6 +66,8 @@ scriptText:
     fileAnIssue: File an issue with AMP
     recommendation: recommendation
     recommendations: recommendations
+  detailsHeadlineImages: "Images to consider"
```
Comment (caroqliu) [inline pages/content/pixi/index.md:69]: Do these need quote marks here?

**Review by caroqliu — APPROVED** [review]
LGTM! Let's also give the `` thumbnails a localized `alt="Thumbnail"`

**Pull request merged**